app/employee-app/backend/service.bal

```diff
@@ -0,0 +1,753 @@
+                return <http:BadRequest>{
+                    body: {
+                        message: "Agreement end date should be after the date of join!"
+                    }
+                };
+            }
+        }
+
+        int employeeId = check database:addRecruit(recruit, userInfo.email);
+        return <http:Created>{
+            body: {
+                message: string `Successfully added the recruit data. Id : ${employeeId.toString()}!`
+            }
+        };
+    }
+
+    # Update recruit data.
+    #
+    # + recruitId - Recruit id of the recruit data to be updated
+    # + recruit - Recruit data to be updated
+    # + return - Recruit update successful or error response
+    resource function patch recruits/[int recruitId](http:RequestContext ctx, types:UpdateRecruitPayload recruit)
+        returns http:Ok|http:Forbidden|error {
+
+        authorization:CustomJwtPayload userInfo = check ctx.getWithType(authorization:HEADER_USER_INFO);
+        if !authorization:checkRoles([authorization:authorizedRoles.recruitmentTeamRole],
+                userInfo.groups) {
+            return <http:Forbidden>{body: {message: "Insufficient privileges!"}};
+        }
+
+        _ = check database:updateRecruit({...recruit, recruitId}, userInfo.email);
+        return <http:Ok>{
+            body: {
+                message: string `Successfully updated the recruits data. Id : ${recruitId}`
+            }
+        };
+    }
+
+    # Skip offer letter.
+    #
+    # + recruitId - ID of the recruit to skip the offer
+    # + return - Skip offer letter successful or error response
+    resource function post recruits/[int recruitId]/offer/skip(http:RequestContext ctx)
+        returns http:Ok|http:Forbidden|error {
+
+        authorization:CustomJwtPayload userInfo = check ctx.getWithType(authorization:HEADER_USER_INFO);
+        if !authorization:checkRoles([authorization:authorizedRoles.recruitmentTeamRole],
+                userInfo.groups) {
+            return <http:Forbidden>{body: {message: "Insufficient privileges!"}};
+        }
+
+        types:Recruit recruit = check database:getRecruit(recruitId);
+        if recruit.status != database:SELECTED {
+            return error("The recruit is not in the selected status to skip the offer letter!");
+        }
+        _ = check database:updateRecruit({status: database:OFFER_SENT, recruitId}, userInfo.email);
+        return <http:Ok>{
+            body: {
+                message: string `Successfully skipped the offer letter for the recruit. Id : ${recruitId}`
+            }
+        };
+    }
+
+    # Sending the offer letter.
+    #
+    # + recruitId - ID of the recruit to send the offer
+    # + return - Offer sent successfully or error response
+    resource function post recruits/[int recruitId]/offer/send(http:RequestContext ctx)
+        returns http:BadRequest|http:Forbidden|http:Ok|http:InternalServerError|http:NotFound|error {
```

Comment:
Best practice is to avoid generic `error` returns from resources.

Reply:
ACK